ecs-cli/modules/cli/local/converter/converter.go

```diff
@@ -175,6 +179,14 @@ func createComposeServices(taskDefinition *ecs.TaskDefinition, metadata *LocalCr
 	return services, nil
 }
 
+func resolveCredentials(taskRoleARN string, useRole bool) string {
+    credsName := endpointsTempCredsPath
+    if parsedRoleARN, err := arnparser.Parse(taskRoleARN); useRole && taskRoleARN != "" && err == nil {
+        credsName = "/" + parsedRoleARN.Resource // The parsed resource is formatted as "role/{roleName}"
+    }
+    return credsName
```

Comment:
We can leverage [guard clauses](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html) to make the conditional easier to parse:
```go
if !useRole {
  return endpointsTempCredsPath
}
parsedARN, err := arnparser.Parse(taskRoleARN)
if err != nil {
  return endpointsTempCredsPath
}
return "/" + parsedARN.Resource
```